File: SingleAPIExample/python/Update.py
import splitit

username, password, api_key = (
    "username",
    "password",
    "api_key",
)

api_client = splitit.ApiClient(splitit.Configuration(username, password, api_key, sandbox=True))


def update_installment_plan(ipn):
    resp = api_client.InstallmentPlanApi.installment_plan_update(
        splitit.UpdateInstallmentPlanRequest(
            installment_plan_number=ipn,
            plan_data=splitit.PlanData(
                ref_order_number="abc123",
                extended_params={
                    "key1": "value1",
                    "key2": "value2",
                }
            )
        )
    )
    return resp

File: SingleAPIExample/java/Update.java
import com.splitit.*;
import com.splitit.sdk.model.*;
import java.math.BigDecimal;


public class Update {
    private String apiKey, username, password;
    public Update(String username, String password, String apiKey){
        this.username = username;
        this.password = password;
        this.apiKey = apiKey;
    }

    public void example(String ipn) throws ApiException{
        ApiClient apiClient = new ApiClient()
            .sandbox(true)
            .username(username)
            .password(password)
            .apiKey(apiKey);


        var result = apiClient.getInstallmentPlanApi().installmentPlanUpdate(
            new UpdateInstallmentPlanRequest()
            .installmentPlanNumber(ipn)
            .planData(
                new PlanData()
                .refOrderNumber("abc123")
                .putExtendedParamsItem("key1", "value1")
                .putExtendedParamsItem("key2", "value2")
            )
        );
        System.out.println(result);
    }
}
